model/patch_lifecycle.go

```diff
@@ -384,3 +384,22 @@ func CancelPatch(p *patch.Patch, caller string) error {
 
 	return errors.WithStack(patch.Remove(patch.ById(p.Id)))
 }
+
+// CancelPatchesWithGithubPatchData runs CancelPatch on patches created before
+// the given time, with the same github author, pr number, and base repository
+func CancelPatchesWithGithubPatchData(createdBefore time.Time, owner, repo string, prNumber int) error {
+	patches, err := patch.Find(patch.ByGithubPRAndCreatedBefore(createdBefore, owner, repo, prNumber))
+	if err != nil {
+		return errors.Wrap(err, "initial patch fetch failed")
+	}
+
+	for i, _ := range patches {
+		if patches[i].Version != "" {
+			if err = CancelPatch(&patches[i], "github-pr"); err != nil {
+				return errors.Wrap(err, "patch cancellation failed")
```

Comment:
Should this add the error to a catcher (`grip.NewSimpleCatcher()`) and continue instead of erring early?

Reply:
agree.

Reply:
if CancelPatch fails, it's going fail due to database communication errors, so if it fails once, it's likely to fail multiple times. Do you still think we should do the catcher?

Side note, what is the difference between NewSimpleCatcher and NewBasicCatcher?